### PR TITLE
Hide recent changes to anonymous user

### PR DESCRIPTION
**Pull request by jhrdt**
DummyUser is a AnonymousUser, so hiding the recent changes will
only affect users that are not logged-in.
Admin will work as usual.

This solution is Django friendly due to the fact that no
code is replace, but rather extented in a Django-way.

### CODE REVIEW

**Comment by jhrdt** [comment]
Die PR is related to #19 .

**Comment by cuducos** [comment]
Hi @jhrdt, many thanks for your contribution.

I do agree with your proposed change, but with one minor comment: _o hiding the recent changes will only affect users that are not logged-in_. This makes little sense in the context of this project as every user will be using the admin without login. This is the whole purpose of this package.

Code-wise it means we don't `if not request.user.is_anonymous` in the template bit, because this template will be only used with `public_admin` Django app, which does not allow logged-in users. Would you mind changing that?

**Comment by jhrdt** [comment]
Hej,

I can do that, but you are aware that this change (your proposed change) will remove the recent changes from `/admin` *public_admin* is placed on top in the settings order, so this `index.html` template will be loaded in `/admin`.

**Comment by cuducos** [comment]
Yes, that’s a great point.

Thus, maybe, we could review other templates in this package to make all them “compatible” with both the native admin and this public admin. What do you think?

**Comment by jhrdt** [comment]
Hi,

then we should keep `if not request.user.is_anonymous` in the template.
This will hide it in the _public admin_ (accessing as dummyuser who is anonymous) and keep the `/admin` intact.

Shall we revert to e97aee6ec363666fb88d476cd0c5cf24ff7a5845?

We could apply a similar logic to the `base.html`.

**Comment by cuducos** [comment]
Yep, sorry if I wasn't clear!
Let's go step by step:

> then we should keep `if not request.user.is_anonymous` in the template.
> This will hide it in the _public admin_ (accessing as dummyuser who is anonymous) and keep the `/admin` intact.

Yes, you have a great point so let's keep it.

> Shall we revert to e97aee6?

Yes, but also feel free to rebase and push with `-f` to organize the branch as you wish : )

> We could apply a similar logic to the `base.html`.

Yes, that was my point, so the template files offered in this package are all following the same principle.

Also, we can review [the docs on these template files](https://github.com/cuducos/django-public-admin/blob/master/docs/usage.rst#templates) to clarify what the new scenario is: that is to say, be clear that the templates here should not change the default behavior of the native admin ; )

**Comment by jhrdt** [comment]
I reverted the changes and applied the same logic to `base.html` .

File: public_admin/templates/admin/base.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {% extends "admin/base.html" %}
 
-{% block welcome-msg %}{% endblock %}
-{% block userlinks %}{% endblock %}
+{% block welcome-msg %}% if not request.user.is_anonymous %}{{ block.super }}{% endif %}{% endblock %}
```
Comment (cuducos) [inline public_admin/templates/admin/base.html]: I believe that inside the `welcome-msg` block, and inside the `if` you added, we have to add [the code from the original admin template](https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/base.html#L38-L41), don't we?

File: public_admin/templates/admin/base.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
+{% block userlinks %}% if not request.user.is_anonymous %}{{ block.super }}{% endif %}{% endblock %}
```
Comment (cuducos) [inline public_admin/templates/admin/base.html]: I believe that inside the `userlinks` block, and inside the `if` you added, we have to add [the code from the original admin template](https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/base.html#L42-L56)?

File: public_admin/templates/admin/base.html (hunk 1 above)
Comment (jhrdt) [reply public_admin/templates/admin/base.html]: `{{ block.super }}` does that, isn't it?

Comment (cuducos) [reply public_admin/templates/admin/base.html]: Ah… true story. I was doing something stupid which was reading code on the mobile, and missed that. 100% my bad. Sorry, mate!

**Comment by anapaulagomes** [comment]
Hi @jhrdt! Thanks to your contribution now our page is much cleaner and less confusing for the users. Thank you :) http://www.dadosabertosdefeira.com.br/painel/

**Pull request merged**